pkg/controller/azdrivernode.go

```diff
@@ -82,6 +87,27 @@ func (r *reconcileAzDriverNode) Reconcile(ctx context.Context, request reconcile
 	return reconcile.Result{Requeue: true}, err
 }
 
+func (r *reconcileAzDriverNode) MarkDetachRequired(ctx context.Context, nodeName string) error {
+	attachments, err := r.azVolumeClient.DiskV1alpha1().AzVolumeAttachments(r.namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("failed to retrieve list of azVolumeAttachment objects for node %s: (%v)", err)
+		return err
+	}
+	for _, attachment := range attachments.Items {
+		if attachment.Status != nil && attachment.Status.AttachmentState != v1alpha1.DetachRequired {
+			updated := attachment.DeepCopy()
+			updated.Status = attachment.Status.DeepCopy()
+			updated.Status.AttachmentState = v1alpha1.DetachRequired
+
+			if _, err := r.azVolumeClient.DiskV1alpha1().AzVolumeAttachments(r.namespace).UpdateStatus(ctx, updated, metav1.UpdateOptions{}); err != nil {
```

Comment:
Rather than set a `DetachRequired` status, you should simply delete the `AzVolumeAttachment` instance. The `AzVolumeAttachment` controller should have added a finalizer to the instance, a so will receive notification of the delete. This should be the trigger to perform the detach.

Reply:
I have added a finalizer logic and removed `AttachmentState` as a whole. `AttachmentState` is no longer needed as if `azVolumeAttachment.Status != nil`, it would mean that it is attached and vice versa.